src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicDetectPreviewFeatureAnalyzer.vb

```diff
@@ -176,6 +176,46 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
         End Function
 
         Protected Overrides Function GetConstraintSyntaxNodeForTypeConstrainedByPreviewTypes(typeOrMethodSymbol As ISymbol, previewInterfaceConstraintSymbol As ISymbol) As SyntaxNode
+            Dim typeSymbolDeclaringReferences = typeOrMethodSymbol.DeclaringSyntaxReferences
+
+            For Each syntaxReference In typeSymbolDeclaringReferences
+                Dim classStatement = TryCast(syntaxReference.GetSyntax(), ClassStatementSyntax)
+                If classStatement IsNot Nothing AndAlso classStatement.TypeParameterList IsNot Nothing Then
+                    Return GetSyntaxNodeFromTypeConstraints(classStatement.TypeParameterList, previewInterfaceConstraintSymbol)
+                End If
+
+                Dim methodDeclaration = TryCast(syntaxReference.GetSyntax(), MethodStatementSyntax)
+                If methodDeclaration IsNot Nothing AndAlso methodDeclaration.TypeParameterList IsNot Nothing Then
+                    Return GetSyntaxNodeFromTypeConstraints(methodDeclaration.TypeParameterList, previewInterfaceConstraintSymbol)
+                End If
+            Next
+            Return Nothing
+        End Function
+
+        Private Function GetSyntaxNodeFromTypeConstraints(typeParameters As TypeParameterListSyntax, previewSymbol As ISymbol) As SyntaxNode
+            For Each typeParameter In typeParameters.Parameters
+                Dim singleConstraint = TryCast(typeParameter.TypeParameterConstraintClause, TypeParameterSingleConstraintClauseSyntax)
+                If singleConstraint IsNot Nothing Then
+                    Return GetTypeConstraints(singleConstraint.Constraint, previewSymbol)
+                End If
+
+                Dim multipleConstraint = TryCast(typeParameter.TypeParameterConstraintClause, TypeParameterMultipleConstraintClauseSyntax)
+                If multipleConstraint IsNot Nothing Then
+                    For Each constriant In multipleConstraint.Constraints
+                        Return GetTypeConstraints(constriant, previewSymbol)
```

Comment:
This is only going to return the first constraint, and will ignore the rest of them, so the test I suggested above is important.

Reply:
good catch!